src/tests.rs

```diff
@@ -0,0 +1,141 @@
+use super::processor;
+use crate::identity;
+use crate::identity::Identity;
+use crate::cache::memory::MemCache;
+use crate::cache::Cache;
+use crate::redis;
+use crate::storage::{RedisStorage};
+use crate::proxy::ProxyWorker;
+use crate::http_workers::HttpWorker;
+
+use crate::twin::{SubstrateTwinDB, Twin};
+
+use crate::anyhow::{Context, Result};
+use crate::twin::TwinDB;
+use crate::http_api::HttpApi;
+
+
+async fn start_rmb(db:SubstrateTwinDB::<MemCache<Twin>>, storage:RedisStorage, ident:identity::Ed25519Signer, address:String) -> Result<()> {
```

Comment:
Why are u giving concrete types, not Traits ? your function doesn't need to know about the concrete types. U just need to accept types that implements certain traits